src/API/Site/Controllers/MerchantCenter/PriceBenchmarksController.php

```diff
@@ -150,11 +189,16 @@ protected function map_price_benchmarks_response( array $benchmark_data, array $
 			}
 		}
 
+		foreach ( $merchant_report_data as $merchant_report_result ) {
+			$mapped_data[ $product_id ]['merchant_report'] = $merchant_report_result['metrics'];
+		}
```

Comment:
This doesn't seem right to me. This method assumes we're getting a set of results for several different products and are making sure we map the results from different sources to the same set of product IDs. This would just add the same metrics to the last $product_id that was mapped previously. We should probably have product ID returned in the metrics query as `segments.program_id`, which we could use to map the report to the request.

Reply:
Yeah, that makes sense to me. So we would only include the metrics data when a the `id` argument is added to the request, not for all products returned in a default request to that endpoint.